src/core/scope.ml

```diff
@@ -105,17 +106,23 @@ let rec get_implicitness : p_term -> bool list = fun t ->
   | P_Wrap(t)    -> get_implicitness t
   | _            -> []
 
-(** [get_args t] decomposes the parser level term [t] into a spine [(h,args)],
-    when [h] is the term at the head of the application and [args] is the list
-    of all its arguments. Note that sugared applications (e.g., infix symbols)
-    are not expanded, so [h] may still be unsugared to an application. *)
-let get_args : p_term -> p_term * p_term list =
+(** [get_parsed_args t] decomposes the parser level term [t] into a spine
+    [(h,args)], when [h] is the term at the head of the application and [args]
+    is the list of all its arguments with infix and prefix operators added
+    using {!module:Pratt}.  Note that sugared applications (e.g., infix
+    symbols) are not expanded, so [h] may still be unsugared to an
+    application. *)
+(* REVIEW: this function is one of the few that use the Pratt parser, and the
+   term is converted from appl to list in [Pratt.parse], then rebuilt into
+   appl node (still by Pratt.parse), then again decomposed into a list by the
+   function. We may make [Pratt.parse] to return already a list of terms. *)
```

Comment:
Yes, why not indeed? Concerning the name, why not "get_pratt_args" instead, to avoid confusion?

Reply:
It's just not done yet, there is no reason.

Reply:
In fact, to do this properly, I think we should use n-ary applications in `p_terms` and binary in `terms`. What do you think?

Reply:
As this implies many changes, I suggest to leave this for another PR.

Reply:
Just, would you mind to rename "get_parsed_args" into, say, "get_pratt_args"?